src/test/java/battery/BatteryParameterStreamerTest.java

```diff
@@ -0,0 +1,127 @@
+package battery;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mockito;
+
+import converter.IConverter;
+import factory.BatteryManagementObjectFactory;
+import junit.framework.Assert;
+import reader.ISourceReader;
+import streamer.ConsoleStreamer;
+import streamer.IStreamer;
+
+public class BatteryParameterStreamerTest {
+
+	private static final File JSON_FILE = new File("resources/BatteryParamValues.json");
+	private static final String CONVERTER_KEY = "CONVERTER";
+	private static final String READER_KEY = "READER";
+	private static final String STREAMER_KEY=  "STREAMER";
+	
+	
+
+	@Rule
+	public ExpectedException exceptionRule = ExpectedException.none();
+
+	@Test
+	public void givenSource_ReadValues_assertNotEmpty() throws Exception {
+		ISourceReader reader = getReader();
+		BatteryParameter batteryParam = (BatteryParameter) reader.readValues(JSON_FILE);
+		org.junit.Assert.assertNotNull(batteryParam);
+		org.junit.Assert.assertTrue(batteryParam.getBatteryParamValueList().size() > 0);
+	}
+
+	@Test
+	public void givenSource_ThrowExceptionForNullFile() throws Exception {
+		exceptionRule.expect(Exception.class);
+		exceptionRule.expectMessage("Invalid File");
+		ISourceReader reader = getReader();
+		reader.readValues(null);
+	}
+
+	@Test
+	public void givenSource_ThrowExceptionForInvalidFile() throws Exception {
+		exceptionRule.expect(Exception.class);
+		exceptionRule.expectMessage("Invalid File");
+		ISourceReader reader = getReader();
+		reader.readValues(new File("D:\\invalidfile.json"));
+	}
+
+	
+	@Test
+	public void givenSource_ValueListConvertToCSVFormat_assertNotEmpty() throws Exception {
+		BatteryParameter batteryParam = getBatteryParameter();
+		IConverter converter = getConverter();
+		List<String> csvList = (List<String>) converter.convert(batteryParam);
+		org.junit.Assert.assertNotNull(csvList);
+		org.junit.Assert.assertTrue(csvList.size() > 0);
```

Comment:
It would be better to assert the content of a few sample rows in csvList. This is to ensure that any mistakes in the output are caught locally and dont need a receiver integration to find out.